api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -92,6 +98,48 @@ public synchronized Order saveOrder(Order order, OrderContext orderContext) thro
 	}
 
 	/**
+	 * @see org.openmrs.api.OrderService#saveOrders(Set)
+	 */
+	public Set<Order> saveOrders(Set<Order> orders) {
+		Set<OrderGroup> orderGroups = new LinkedHashSet<OrderGroup>();
+		for (Order order : orders) {
+			if (order.getOrderId() == null) {
+				if (order.getOrderGroup() != null) {
+					order.setOrderGroup(getOrderGroup(orderGroups, order.getOrderGroup()));
+					setSequenceNumberForOrders(order);
+				}
+				Context.getOrderService().saveOrder(order, null);
+			}
+		}
+		return orders;
+	}
+
+
+
+	private OrderGroup getOrderGroup(Set<OrderGroup> orderGroups, OrderGroup newOrderGroup) {
```

Comment:
I think this should be named something like createOrderGroupIfNecessary for clarity purposes, it would also be nice to have some comments about what's going on, it is helpful for some else reading this code